apps/emqx/src/emqx_shared_sub.erl

```diff
@@ -72,20 +73,29 @@
     init_monitors/0
 ]).
 
+%% Client session lifecycle hooks
+-export([
+    on_client_session_terminated/4
+]).
+
 -export_type([strategy/0]).
 
 -type strategy() ::
     random
     | round_robin
     | round_robin_per_group
     | sticky
+    | sticky_clientid
+    | sticky_clientid_leastpubs
     | local
     | hash_clientid
     | hash_topic.
 
 -define(SERVER, ?MODULE).
 -define(TAB, emqx_shared_subscription).
 -define(SHARED_SUBS_ROUND_ROBIN_COUNTER, emqx_shared_subscriber_round_robin_counter).
+-define(SHARED_SUBS_LEASTPUBS_COUNTER, emqx_shared_subscriber_least_pubs_counter).
```

Comment:
please help to refactor the code to use one ETS table for both.
e.g. rename `emqx_shared_subscriber_round_robin_counter` to `emqx_shared_subscriber_dispatch_counter`.